tests/queries/0_stateless/02900_date_time_check_overflow.sql

```diff
@@ -0,0 +1,50 @@
+SET session_timezone = 'UTC';
+
+SELECT 'ignore';
+SET date_time_overflow_mode = 'ignore';
+SELECT toDateTime(toDateTime64('1900-01-01 00:00:00.123', 3));
+SELECT toDateTime(toDateTime64('2299-12-31 23:59:59.999', 3));
+
+SELECT toDateTime(toDate32('1900-01-01'));
+SELECT toDateTime(toDate32('2299-12-31'));
+
+SELECT toDateTime(toDate('2149-06-06'));
+
+SELECT toDate(toDateTime64('1900-01-01 00:00:00.123', 3));
+SELECT toDate(toDateTime64('2149-06-07 00:00:00.123', 3));
+SELECT toDate(toDateTime64('2299-12-31 23:59:59.999', 3));
+
+SELECT toDate(toDate32('1900-01-01'));
+SELECT toDate(toDate32('2299-12-31'));
+
+
+SELECT 'No output on `throw`';
+SET date_time_overflow_mode = 'throw';
+SELECT toDateTime(toDateTime64('1900-01-01 00:00:00.123', 3)); -- { serverError VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE }
+SELECT toDateTime(toDateTime64('2299-12-31 23:59:59.999', 3)); -- { serverError VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE }
+SELECT toDateTime(toDate32('1900-01-01')); -- { serverError VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE }
+SELECT toDateTime(toDate32('2299-12-31')); -- { serverError VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE }
+SELECT toDateTime(toDate('2149-06-06')); -- { serverError VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE }
```

Comment:
`2149-06-06` is outside the range of `Date`, right? So the test could just be `SELECT toDate('2149-06-06');`? In that case, it would be good to have also tests for standalone construction of Date32, DateTime and DateTime64 with string values outside of their range. And what about tests with integer arguments, e.g. `SELECT toDate(300000000000)`?

Reply:
> So the test could just be SELECT toDate('2149-06-06');

No, it is a bit different thing. This is about parsing, not conversion

Reply:
And what about funny things like conversion from floats?

Reply:
Ah, now I see, `SELECT toDateTime(toDate('2149-06-06'));` tests Date-to-DateTime which have different upper boundaries. Okay then.